### DISCUSSION TITLE
useWhisper hook with Real-Time transcription for React.js

### DISCUSSION THREAD

**Discussion by chengsokdara** [openai/whisper#1042] [Show and tell]
Hello I just published a React library for anyone who want to implement Speech-to-Text using OpenAI Whisper in their React.js web app.

Try it:
`yarn add @chengsokdara/use-whisper`
`npm i @chengsokdara/use-whisper`

Repo:
[https://github.com/chengsokdara/use-whisper](https://github.com/chengsokdara/use-whisper)

Demo: (Whisper seems to can not understand my accent 😅)

https://user-images.githubusercontent.com/2707253/224465842-991a989c-ea73-49aa-8e6d-e3b520ee0065.mp4

Guide:
[Medium - Transform Your Speech into Text with the Power of OpenAI and useWhisper](https://medium.com/@chengsokdara/transform-your-speech-into-text-with-the-power-of-openai-and-usewhisper-ca78f8e74160)

**Comment by ThantHtetAung101** [comment]
Is there a way to use whisper ai for free? I mean whisper ai free but using it with openAI api costs money. I know it's becuz of gpu calculation from cloud. I think we can use google colab instead of openAI. But i'm kind of worried about delay time between the mic and the output transcription. Is there a way to use whisper ai free in react projects?

**Reply by brian316** [reply]
use on your own pc. or try a free trial from online cloud vendors like Google, AWS, linode

**Comment by Usama9999** [comment]
Are you using your mic for speaking or using some predefined audio?
